include/boost/url/detail/encode.hpp

```diff
@@ -419,8 +419,21 @@ re_encode_unchecked(
     return dest - dest0 - dn;
 }
 
+template<
+    class CharSet =
+        grammar::all_chars_t>
```

Comment:
the implementation signature should not have a defaulted template argument

Reply:
OK. I left the template argument in `url/encode.hpp` only